src/js/components/Topic/Index.jsx

```diff
@@ -33,19 +35,32 @@ export default class Index extends React.Component {
         clearTimeout(this.timer);
 
         this.timer = setTimeout(() => {
-            this.setState({
-                search: value
-            });
+            this.props.location.query.search = value;
```

Comment:
i think, you shouldn't modify the props or location. you should clone it before.